servant-foreign/src/Servant/Foreign/Internal.hs

```diff
@@ -79,47 +80,59 @@ data ArgType
 
 makePrisms ''ArgType
 
-data QueryArg = QueryArg
-  { _argName :: Arg
-  , _argType :: ArgType
-  } deriving (Eq, Show)
+data QueryArg f = QueryArg
+  { _queryArgName :: Arg f
+  , _queryArgType :: ArgType
+  }
+
+deriving instance Eq f => Eq (QueryArg f)
+deriving instance Show f => Show (QueryArg f)
 
 makeLenses ''QueryArg
 
-data HeaderArg = HeaderArg
-  { _headerArg :: Arg }
+data HeaderArg f = HeaderArg
+  { _headerArg :: Arg f }
   | ReplaceHeaderArg
-  { _headerArg     :: Arg
+  { _headerArg     :: Arg f
   , _headerPattern :: Text
-  } deriving (Eq, Show)
+  }
+
+deriving instance Eq f => Eq (HeaderArg f)
+deriving instance Show f => Show (HeaderArg f)
 
 makeLenses ''HeaderArg
 
 makePrisms ''HeaderArg
 
-data Url = Url
-  { _path     :: Path
-  , _queryStr :: [QueryArg]
-  } deriving (Eq, Show)
+data Url f = Url
+  { _path     :: Path f
+  , _queryStr :: [QueryArg f]
+  }
+
+deriving instance Eq f => Eq (Url f)
+deriving instance Show f => Show (Url f)
 
-defUrl :: Url
+defUrl :: Url f
 defUrl = Url [] []
 
 makeLenses ''Url
 
-data Req = Req
-  { _reqUrl        :: Url
+data Req f = Req
+  { _reqUrl        :: Url f
   , _reqMethod     :: HTTP.Method
-  , _reqHeaders    :: [HeaderArg]
-  , _reqBody       :: Maybe ForeignType
-  , _reqReturnType :: ForeignType
+  , _reqHeaders    :: [HeaderArg f]
+  , _reqBody       :: Maybe f
+  , _reqReturnType :: Maybe f
```

Comment:
On reflection, previously it was getting set to `""` by `defReq`, which is fairly equally odd.